### PR TITLE
Refactor testruns search view to class-based

### PR DESCRIPTION
**Pull request by RMadjev**

### CODE REVIEW

File: tcms/testruns/tests/test_views.py [hunk 1]
```diff
@@ -334,6 +334,11 @@ def test_search_page_is_shown(self):
         response = self.client.get(self.search_runs_url)
         self.assertContains(response, '<input id="id_summary" type="text"')
 
+    def test_search_page_is_shown_with_get_parameter_used(self):
+        url = self.search_runs_url + '?product=1'
+        response = self.client.get(url)
+        self.assertContains(response, '<option value="1" selected>Kiwi</option>')
```
Comment (atodorov) [inline tcms/testruns/tests/test_views.py]: Use ProductFactory to create a new product in setUpTestData and assert if it exists and is selected. 

There is not product named Kiwi b/c it has not been created and this is why test is failing.

Use full project name - Kiwi TCMS.

File: tcms/testruns/views.py [hunk 2]
```diff
@@ -122,15 +122,19 @@ def post(self, request):
...
-    return render(request, 'testruns/search.html', context_data)
+    template_name = 'testruns/search.html'
+
+    def get_context_data(self, **kwargs):
+        form = SearchRunForm(self.request.GET)
+        form.populate(product_id=self.request.GET.get('product'))
+
+        context_data = {
+            'form': form,
+        }
```
Comment (atodorov) [inline tcms/testruns/views.py]: You can `return` directly the dict literal here and save a few lines of code/1 unneeded variable. Up to you.

File: tcms/testruns/tests/test_views.py (hunk 1 above)
Comment (RMadjev) [reply tcms/testruns/tests/test_views.py]: The test is passing locally. The product is created in the grandparent class(BasePlanCase). I think at the first push the Scrutinizer was ok.

Comment (atodorov) [reply tcms/testruns/tests/test_views.py]: Then use `self.product` and make sure the HTML you are asserting on is the same. Also try `html=True` parameter to the assert call.

**Pull request merged**